### PR TITLE
Align WatchService events on all platforms/file systems

### LINKED ISSUE #3952: Maven build failing in Eclipse SmartHome Core Tests
With the current master branch on a Windows 10 Pro 64 bits, a "mvn clean install" is failing in Eclipse SmartHome Core Tests with the following errors:
```
Running org.eclipse.smarthome.core.service.AbstractWatchServiceTest
Tests run: 8, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 51.021 sec <<< FAILURE! - in org.eclipse.smarthome.core.service.AbstractWatchServiceTest
AbstractWatchQueueReader processWatchEvent path in subsubdir provided with correct arguments(org.eclipse.smarthome.core.service.AbstractWatchServiceTest)  Time elapsed: 0.625 sec
subdirs are registered and modifications are watched(org.eclipse.smarthome.core.service.AbstractWatchServiceTest)  Time elapsed: 30.081 sec  <<< FAILURE!
java.lang.AssertionError: Exactly 2 watch events were expected, but were: [Watch Event: count 1; kind: ENTRY_CREATE; path: watchDirectory\correctlyWatchedSubDir\correctSubDirInnerFile]
Expected: is <2>
     but: was <1>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.junit.Assert.assertThat(Assert.java:865)
        at org.junit.Assert$assertThat$3.callStatic(Unknown Source)
        at org.eclipse.smarthome.core.service.AbstractWatchServiceTest$_assertAllEventsAreProcessed_closure9.doCall(AbstractWatchServiceTest.groovy:254)
        at org.eclipse.smarthome.core.service.AbstractWatchServiceTest$_assertAllEventsAreProcessed_closure9.doCall(AbstractWatchServiceTest.groovy)
        at sun.reflect.GeneratedMethodAccessor43.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:278)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
        at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
        at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:220)
        at sun.reflect.GeneratedMethodAccessor11.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrap.invoke(PogoMetaMethodSite.java:187)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153)
        at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:191)
        at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:145)
        at org.eclipse.smarthome.core.service.AbstractWatchServiceTest.assertAllEventsAreProcessed(AbstractWatchServiceTest.groovy:254)

AbstractWatchQueueReader processWatchEvent path in root provided with correct arguments(org.eclipse.smarthome.core.service.AbstractWatchServiceTest)  Time elapsed: 0.578 sec
same file names in root and subdir are correctly processed(org.eclipse.smarthome.core.service.AbstractWatchServiceTest)  Time elapsed: 3.001 sec
AbstractWatchQueueReader processWatchEvent path in subdir provided with correct arguments(org.eclipse.smarthome.core.service.AbstractWatchServiceTest)  Time elapsed: 0.64 sec
subdirs are not registered, but dirs modifications are watched(org.eclipse.smarthome.core.service.AbstractWatchServiceTest)  Time elapsed: 10.064 sec  <<< FAILURE!
java.lang.AssertionError: Exactly 1 watch events were expected, but were: []
Expected: is <1>
     but: was <0>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.junit.Assert.assertThat(Assert.java:865)
        at org.junit.Assert$assertThat$3.callStatic(Unknown Source)
        at org.eclipse.smarthome.core.service.AbstractWatchServiceTest$_assertDirectoryModifyEventIsProcessed_closure12.doCall(AbstractWatchServiceTest.groovy:300)
        at org.eclipse.smarthome.core.service.AbstractWatchServiceTest$_assertDirectoryModifyEventIsProcessed_closure12.doCall(AbstractWatchServiceTest.groovy)
        at sun.reflect.GeneratedMethodAccessor47.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:278)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
        at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
        at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:54)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
        at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:220)
        at sun.reflect.GeneratedMethodAccessor11.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrap.invoke(PogoMetaMethodSite.java:187)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153)
        at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:191)
        at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy)
        at sun.reflect.GeneratedMethodAccessor10.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
        at org.eclipse.smarthome.core.service.AbstractWatchServiceTest.assertDirectoryModifyEventIsProcessed(AbstractWatchServiceTest.groovy:300)

subdirs are not registered and modifications are not watched(org.eclipse.smarthome.core.service.AbstractWatchServiceTest)  Time elapsed: 3 sec
subdirs are registered, but dirs modifications are not watched(org.eclipse.smarthome.core.service.AbstractWatchServiceTest)  Time elapsed: 3.016 sec
Running org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.078 sec - in org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest
assert two items with same name can not be added(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.015 sec
assert items are there once the factory gets added(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.016 sec
assert items are added as group members on deferred creation(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.016 sec
assert group functions are stored and retrieved as well(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0 sec
assert getItems returns item from registered ManagedItemProvider(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0 sec
assert removal returns old value(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0 sec
assert tags are stored and retrieve as well(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0 sec
updating existing item returns old value(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0 sec
assert remove recursively works(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.015 sec
assert functions are stored and retrieved as well(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0 sec

Results :

Failed tests:
  AbstractWatchServiceTest.subdirs are registered and modifications are watched:164->assertAllEventsAreProcessed:254->OSGiTest.waitForAssert:-1->OSGiTest.waitForAssert:191->OSGiTest.waitForAssert:220 Exactly 2 watch events were expected, but were: [Watch Event: count 1; kind: ENTRY_CREATE; path: watchDirectory\correctlyWatchedSubDir\correctSubDirInnerFile]
Expected: is <2>
     but: was <1>
  AbstractWatchServiceTest.subdirs are not registered, but dirs modifications are watched:234->assertDirectoryModifyEventIsProcessed:300->OSGiTest.waitForAssert:-1->OSGiTest.waitForAssert:191->OSGiTest.waitForAssert:220 Exactly 1 watch events were expected, but were: []
Expected: is <1>
     but: was <0>

Tests run: 289, Failures: 2, Errors: 0, Skipped: 1
```

### PR DESCRIPTION
**Pull request by sjka**
Apparently some platforms signal for every file creation/deletion
a modification of the containing directory while others dont.

As this appears to be not only specific to some OS but also
depends on the file system used, such directory modifications
are suppressed now completely.

So instead of adding yet another case to the test I rather
streamlined the behavior of the productive code to behave
consistently (and thereby ported the test-case to Java).

It's labeled as "APIBreaking" because a method was dropped from AbstractWatchService, which wasn't overridden though by any of the implementations. No bindings inherit from it and I wouldn't expect any additional subclasses of it outside of the framework.

fixes #3952
Signed-off-by: Simon Kaufmann <simon.kfm@googlemail.com>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Sounds as if Java fails to provide a reliable abstraction from the OS on this topic...
I honestly have no clue whether the changes of this PR are doing the right thing - but I see that you at least still have to convince one unit test :-)

**Comment by sjka** [comment]
Yepp - looks like I know hit the famous file-is-empty-for-a-short-moment situation on Linux 😩

File: bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/XmlDocumentBundleTracker.java [hunk 1]
```diff
@@ -58,7 +58,7 @@
  */
 public class XmlDocumentBundleTracker<T> extends BundleTracker<Bundle> {
 
-    public static final String THREAD_POOL_NAME = "file-processing";
+    public static final String THREAD_POOL_NAME = "xml-processing";
```
Comment (kaikreuzer) [inline bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/XmlDocumentBundleTracker.java]: I'd assume that you didn't want to revert this change with https://github.com/eclipse/smarthome/pull/4851/commits/ecb254c22bf15b00613f8f6c138a47047b815f88.

File: bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/folder/FolderObserver.java [hunk 2]
```diff
@@ -227,31 +219,39 @@ public boolean accept(File dir, String name) {
         }
     }
 
+    // TODO remove once #3562 got resolved
```
Comment (kaikreuzer) [inline bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/folder/FolderObserver.java]: I'd assume that you didn't want to revert this change with https://github.com/eclipse/smarthome/pull/4851/commits/ecb254c22bf15b00613f8f6c138a47047b815f88.

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/service/WatchQueueReader.java [hunk 3]
```diff
@@ -251,6 +282,55 @@ public void run() {
         }
     }
 
+    /**
+     * Schedules forwarding of the event to the listeners (if appliccable).
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/service/WatchQueueReader.java]: typo: applicable
